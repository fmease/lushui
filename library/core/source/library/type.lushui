; The type of types and related functionality.
module

; The type of types.
@public @intrinsic
data Type
