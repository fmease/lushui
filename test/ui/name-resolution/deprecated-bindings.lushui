;;; TEST pass check

use extern.core.type.Type

@deprecated
data Foo: Type of

@deprecated
Bar: Type = Foo

@(deprecated (reason "it was not the right thing"))
Baz: Type -> Type = for (A: Type) => Type

X: Type = Baz Bar
