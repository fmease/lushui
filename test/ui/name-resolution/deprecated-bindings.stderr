warning[deprecated]: use of deprecated binding ‘topmost.Foo’
  ┌─ ${TEST_FOLDER}/name-resolution/deprecated-bindings.lushui:9:13
  │
9 │ Bar: Type = Foo
  │             ═══

warning[deprecated]: use of deprecated binding ‘topmost.Baz’: it was not the right thing
   ┌─ ${TEST_FOLDER}/name-resolution/deprecated-bindings.lushui:14:11
   │
14 │ X: Type = Baz Bar
   │           ═══

warning[deprecated]: use of deprecated binding ‘topmost.Bar’
   ┌─ ${TEST_FOLDER}/name-resolution/deprecated-bindings.lushui:14:15
   │
14 │ X: Type = Baz Bar
   │               ═══

warning: emitted 3 warnings

